backend/app/services/summary_service.py

```
@@ -0,0 +1,505 @@
+
+                    # Check content length, truncate if too long
+                    max_length = 50000  # About 50k characters
+                    if len(content) > max_length:
+                        logger.warning(
+                            f"[SummaryService] Document too long ({len(content)} chars), truncating"
+                        )
+                        content = content[:max_length] + "\n\n[Content truncated...]"
+
+                    return content
+
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)